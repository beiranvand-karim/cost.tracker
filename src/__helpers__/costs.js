 const costs = [
   {
      cost: 1000,
      comment: 'this is a comment',
      category: 'CLOTHING'
   },
   {
      cost: 1500,
      comment: 'this is a comment',
      category: 'RENT'
   },
   {
      cost: 2000,
      comment: 'this is a comment',
      category: 'FOOD'
   }
];

export default costs;
