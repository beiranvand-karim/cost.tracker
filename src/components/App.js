import ButtonDropdownButtonContainer from './ButtonDropdownContainer';
import ModalContainer from './ModalContainer';
import {Button} from 'reactstrap';
import React, {useState} from 'react';
import {useSelector} from 'react-redux';
import CostTable from './CostTable';
import SumOfTransactions from './SumOfTransactions';
import '../styles/App.scss';

function App() {

   const [modal, setModal] = useState(false);
   const toggle = () => setModal(!modal);

   const {costs} = useSelector(state => state.costs);

   return (
      <div className="my-container">
         <Button color="primary" className="some-item" onClick={toggle}>add a cost</Button>
         <ButtonDropdownButtonContainer className="some-item"/>
         <SumOfTransactions className="some-item"/>
         <ModalContainer toggle={toggle} modal={modal} setModal={setModal}/>
         <CostTable costs={costs}/>
      </div>
   );
}

export default App;
