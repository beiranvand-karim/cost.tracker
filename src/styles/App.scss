/*

.my-container {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-row-gap: 40px;
}

.some-item {
  grid-column:  2/ span 10;
}
*/
