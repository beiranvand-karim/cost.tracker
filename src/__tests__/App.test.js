/* global expect, describe, it, beforeEach */
import React from "react";
import App from "../components/App";
import {Provider} from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import {shallow} from "enzyme";
import costs from "../__helpers__/costs";

describe('App', () => {

   const mockStore = configureStore();
   let store;

   beforeEach(() => {
      store = mockStore({
         costs: {
            costs
         },
         originalCosts: {
            costs
         }
      });
   });

   it('should render without crashing', () => {
      shallow(<Provider store={store}>
         <App/>
      </Provider>)
   });

   it('should matches the snapshot', () => {
      const tree = renderer.create(<Provider store={store}>
         <App/>
      </Provider>).toJSON();
      expect(tree).toMatchSnapshot()
   });

});
