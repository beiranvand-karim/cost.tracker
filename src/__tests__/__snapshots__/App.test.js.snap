// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should matches the snapshot 1`] = `
<div
  className="my-container"
>
  <button
    aria-label={null}
    className="some-item btn btn-primary"
    onClick={[Function]}
    type="button"
  >
    add a cost
  </button>
  <div>
    <select
      onChange={[Function]}
    >
      <option
        value="NO_FILTER"
      >
        NO_FILTER
      </option>
      <option
        value="FOOD"
      >
        FOOD
      </option>
      <option
        value="CLOTHING"
      >
        CLOTHING
      </option>
      <option
        value="RENT"
      >
        RENT
      </option>
      <option
        value="BILLS"
      >
        BILLS
      </option>
    </select>
  </div>
  <div>
    4500
  </div>
  <div
    className="table-responsive"
  >
    <table
      className="table"
    >
      <thead>
        <tr>
          <th>
            #
          </th>
          <th>
            Cost
          </th>
          <th>
            Category
          </th>
          <th>
            Comment
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            scope="row"
          >
            0
          </th>
          <td>
            1000
          </td>
          <td>
            CLOTHING
          </td>
          <td>
            this is a comment
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            1
          </th>
          <td>
            1500
          </td>
          <td>
            RENT
          </td>
          <td>
            this is a comment
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            2
          </th>
          <td>
            2000
          </td>
          <td>
            FOOD
          </td>
          <td>
            this is a comment
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
